src/main/java/rockset/AvroParser.java

```diff
@@ -56,33 +56,46 @@ private boolean isLogicalType(Schema schema) {
     return LOGICAL_TYPE_CONVERTERS.containsKey(schema.name());
   }
 
-  private Map<String, Object> convertLogicalTypes(Schema valueSchema, Map<String, Object> map) {
+  private Object convertType(Schema schema, Object o) {
+    if (isLogicalType(schema)) {
+      return convertLogicalType(schema, o);
+    }
+
+    Type type = schema.type();
+
+    switch (type) {
+      case STRUCT:
+      case MAP:
+        return convertLogicalTypesMap(schema, (Map<String, Object>)o);
```

Comment:
We should probably have 2 separate methods handling maps and structs. It seems weird that we have a type check here and again in `getSchemaForField` (current called for both structs and maps but should only be invoked for structs ideally). Can come later though

Reply:
yeah, thats a little ugly...thanks, lets address for later...